### ISSUE TITLE
Discuss package/library names

### ISSUE THREAD

**Issue by cwe1ss** [opentracing-contrib/csharp-aspnetcore#2]
This solution currently contains two projects that would ship as libraries on nuget.org:

#### OpenTracing.Contrib.AspNetCore
This project currently contains instrumentation for incoming HTTP requests and for some MVC components (actions, views). It has dependencies on `Microsoft.AspNetCore.*` libraries.

I think this name is pretty self-explanatory and should be ok.

#### OpenTracing.Contrib.Core
This project currently contains instrumentation for outgoing HTTP calls via `HttpClient` and for EntityFramework Core calls. EF Core is included because instrumenting it works without a dependency on it. The idea behind this separate `Core` project is to extract instrumentation that could also be used for non-web projects. However, due to its current architecture with dependencies on `Microsoft.Extensions.*` libraries (DependencyInjection, Logging, etc) it still requires the non-web project to use the "new" `Microsoft.Extensions.*` stack from the ASP.NET Core team. 

Also, due to its dependency on `Microsoft.Extensions.DiagnosticAdapter` it currently only works on .NET Core. From ASP.NET 2.1 it will also support .NET 4.6+ though.

In other words, the `Core` project is a sort-of BCL instrumentation library but it requires the host to use the "new" `Microsoft.Extensions.*` stack and it (currently) only works on .NET Core. 

`OpenTracing.Contrib.Core` was the best I could come up with but I'm open to suggestions here!

**Comment by cwe1ss** [comment]
I did some tests with the full .NET framework and it currently does not contain the same instrumentation possibilities as .NET Core (e.g. DiagnosticSource instrumentation for HTTP calls). Releasing these packages for the full .NET is therefore a non-goal for the foreseeable future.

I will therefore rename the package `OpenTracing.Contrib.Core` to `OpenTracing.Contrib.NetCore` to make it clear that it is for .NET Core. There's many packages on NuGet that [follow this naming scheme](https://www.nuget.org/packages?q=NetCore) so I think this should be ok!
